libraries/app/database_api.cpp

```diff
@@ -2362,22 +2362,24 @@ vector<proposal_object> database_api::get_proposed_transactions( const std::stri
    return my->get_proposed_transactions( account_id_or_name );
 }
 
-/** TODO: add secondary index that will accelerate this process */
 vector<proposal_object> database_api_impl::get_proposed_transactions( const std::string account_id_or_name )const
 {
-   const auto& idx = _db.get_index_type<proposal_index>();
+   const auto& proposal_idx = _db.get_index_type<proposal_index>();
+   const auto& pidx = dynamic_cast<const base_primary_index&>(proposal_idx);
```

Comment:
I think you can combine the two lines above into `... idx = _db.get_index_type< primary_index< proposal_index > >();`

Reply:
Thank you, also used the code to do the same in `get_full_accounts` at https://github.com/bitshares/bitshares-core/pull/1749/commits/66c0f40ae89130ce018ffb4c92a5d588d698c3d9